### PR TITLE
ciao-image: Integrate into ciao-controller service

### PR DESCRIPTION
**Pull request by obedmr**
This PR is addressing issue: #801 

This PR integrates ``ciao-image`` into ``ciao-controller``.
It means that:
- Controller will expose Ciao Image service
- Ciao-image Service will use same controller API certificates
- Ciao-image Service will live along the Compute (Controller) and Storage services

Signed-off-by: Munoz, Obed N <obed.n.munoz@intel.com>

### CODE REVIEW

**Comment by markdryan** [comment]
You will need to make some changes to setup.sh, i.e., remove the call to run_image.sh and the symbolic links for the image service cert

**Comment by obedmr** [comment]
@markdryan, sure, added. I removed run_image.sh references from setup.sh

File: ciao-controller/openstack_image.go [hunk 1]
```diff
@@ -185,7 +188,7 @@ type Config struct {
 	Password string
 }
 
-func getIdentityClient(config Config) (*gophercloud.ServiceClient, error) {
+func getIdentityClient(config ImageConfig) (*gophercloud.ServiceClient, error) {
 	opt := gophercloud.AuthOptions{
```
Comment (kaccardi) [inline ciao-controller/openstack_image.go]: if you move into controller, do you still need to do this? controller already has an identity client - I don't think it needs 2.

File: ciao-controller/openstack_image.go [hunk 2]
```diff
@@ -206,12 +209,50 @@ func getIdentityClient(config Config) (*gophercloud.ServiceClient, error) {
...
 // service.
-func Start(config Config) error {
+func (c *controller) startImageService() error {
+
+	dbDir := filepath.Dir(*imageDatastoreLocation)
+	dbFile := filepath.Base(*imageDatastoreLocation)
+	glog.Infof("Image Datastore Location: %v", *imageDatastoreLocation)
+
+	metaDs := &datastore.MetaDs{
+		DbProvider: database.NewBoltDBProvider(),
```
Comment (kaccardi) [inline ciao-controller/openstack_image.go]: can you alias the image datastore to something other than "datastore"? I think it'll be less confusing for future contributors who might get it and the controller's internal/datastore package confused.

File: ciao-controller/openstack_image.go [hunk 3]
```diff
@@ -206,12 +209,50 @@ func getIdentityClient(config Config) (*gophercloud.ServiceClient, error) {
...
+	if err != nil {
+		glog.Fatalf("Error on DB Tables Initialization:%v ", err)
+	}
+
+	rawDs := &datastore.Posix{
+		MountPoint: c.image.MountPoint,
+	}
+
+	config := ImageConfig{
+		Port:             9292,
```
Comment (kaccardi) [inline ciao-controller/openstack_image.go]: dont hard code the port. needs to come from config.

File: ciao-controller/openstack_image.go (hunk 2 above)
Comment (obedmr) [inline ciao-controller/openstack_image.go]: sure, I'll change it to imageDatastore

File: ciao-controller/openstack_image.go (hunk 1 above)
Comment (obedmr) [inline ciao-controller/openstack_image.go]: @kaccardi, yeah, you're right, not needed, /me removing it

**Comment by obedmr** [comment]
@markdryan, @kaccardi, @rbradford when you have a chance, could you take a look on this.

**Comment by markdryan** [comment]
I tested this on SingleVM today and it seems to be working.  I was even able to get all the image BAT tests, pending in PR #815 to work.  I noticed one small problem with setup.sh script

```
```

**Comment by obedmr** [comment]
@markdryan, thanks, good catch, I also found that I didn't remove the lines that referenced to ciao-image certs, I updated it, I think it should be working now.

**Comment by rbradford** [comment]
Waiting on approval from @kaccardi for merging.

**Pull request merged**